remoting/subsystem/src/main/java/org/jboss/as/remoting/RemotingSubsystemAdd.java

```diff
@@ -72,26 +71,33 @@ protected void performRuntime(OperationContext context, ModelNode operation, Res
         // WFCORE-4510 -- the effective endpoint configuration is from the root subsystem resource,
         // not from the placeholder configuration=endpoint child resource.
         ModelNode endpointModel = resource.getModel();
-        String workerName = WORKER.resolveModelAttribute(context, endpointModel).asString();
-
-        final OptionMap map = EndpointConfigFactory.populate(context, endpointModel);
-
-        // create endpoint
-        final String nodeName = WildFlySecurityManager.getPropertyPrivileged(RemotingExtension.NODE_NAME_PROPERTY, null);
-
-        // In case of a managed server the subsystem endpoint might already be installed {@see DomainServerCommunicationServices}
-        if (context.getProcessType() == ProcessType.DOMAIN_SERVER) {
-            final ServiceController<?> controller = context.getServiceRegistry(false).getService(RemotingServices.SUBSYSTEM_ENDPOINT);
-            if (controller != null) {
-                // if installed, just skip the rest
-                return;
-            }
+        String workerName = WORKER.resolveModelAttribute(context, endpointModel).asStringOrNull();
+
+        CapabilityServiceBuilder<?> builder = context.getCapabilityServiceTarget().addCapability(REMOTING_ENDPOINT_CAPABILITY);
+        Consumer<Endpoint> endpointConsumer = builder.provides(REMOTING_ENDPOINT_CAPABILITY);
+
+        if (workerName != null) {
+            OptionMap map = EndpointConfigFactory.populate(context, endpointModel);
+            String nodeName = WildFlySecurityManager.getPropertyPrivileged(RemotingExtension.NODE_NAME_PROPERTY, null);
+
+            Supplier<XnioWorker> workerSupplier = builder.requiresCapability(IO_WORKER_CAPABILITY_NAME, XnioWorker.class, workerName);
+            builder.setInstance(new EndpointService(endpointConsumer, workerSupplier, nodeName, EndpointService.EndpointType.SUBSYSTEM, map));
+        } else {
+            // Otherwise, reference endpoint created by DomainServerCommunicationServices
```

Comment:
Perhaps

"// This is a domain server as WORKER is a required attribute otherwise. Reference endpoint created by DomainServerCommunicationServices"

When looking at this it took me a while to figure out why the else block just assumed a domain server.

Reply:
I will clarify accordingly.  Thanks for the feedback.